modular_skyrat/master_files/code/modules/pain_shock/stat_handler.dm

```diff
@@ -0,0 +1,58 @@
+/mob/living/carbon/set_stat(new_stat)
+	. = ..()
+	if(should_be_critical(new_stat))
+		ADD_TRAIT(src, TRAIT_HANDS_BLOCKED, STAT_TRAIT)
+		ADD_TRAIT(src, TRAIT_CRITICAL_CONDITION, STAT_TRAIT)
+		if(flow_rate <= 80)
+			ADD_TRAIT(src, TRAIT_IMMOBILIZED, TRAIT_KNOCKEDOUT)
+			ADD_TRAIT(src, TRAIT_FLOORED, STAT_TRAIT)
+		if(stat == DEAD)
+			consider_the_score(new_stat)
+	else
+		switch(new_stat)
+
+			if(CONSCIOUS && !should_be_critical(new_stat))
+				REMOVE_TRAIT(src, TRAIT_HANDS_BLOCKED, STAT_TRAIT)
+				REMOVE_TRAIT(src, TRAIT_INCAPACITATED, STAT_TRAIT)
+				REMOVE_TRAIT(src, TRAIT_FLOORED, STAT_TRAIT)
+				REMOVE_TRAIT(src, TRAIT_CRITICAL_CONDITION, STAT_TRAIT)
+				cure_blind(UNCONSCIOUS_TRAIT)
+				if(stat == DEAD)
+					consider_the_score(new_stat)
+				if(stat >= UNCONSCIOUS)
+					REMOVE_TRAIT(src, TRAIT_FLOORED, STAT_TRAIT)
+					REMOVE_TRAIT(src, TRAIT_IMMOBILIZED, TRAIT_KNOCKEDOUT)
+			if(UNCONSCIOUS)
+				if(stat == DEAD)
+					consider_the_score(new_stat)
+				if(stat != HARD_CRIT)
+					become_blind(UNCONSCIOUS_TRAIT)
+				if(should_be_critical() && !HAS_TRAIT(src, TRAIT_NOSOFTCRIT))
+					ADD_TRAIT(src, TRAIT_FLOORED, STAT_TRAIT)
+					ADD_TRAIT(src, TRAIT_CRITICAL_CONDITION, STAT_TRAIT)
+				else
+					REMOVE_TRAIT(src, TRAIT_CRITICAL_CONDITION, STAT_TRAIT)
+					cure_blind(UNCONSCIOUS_TRAIT)
+			if(DEAD)
+				ADD_TRAIT(src, TRAIT_FLOORED, STAT_TRAIT)
+				REMOVE_TRAIT(src, TRAIT_CRITICAL_CONDITION, STAT_TRAIT)
+				if(stat == DEAD)
+					return
+				consider_the_score(new_stat)
+
+/mob/living/carbon/proc/consider_the_score(new_stat)
+	if(stat == DEAD && new_stat != DEAD)
+		remove_from_dead_mob_list()
+		add_to_alive_mob_list()
+	else
+		remove_from_alive_mob_list()
+		add_to_dead_mob_list()
+/mob/living/carbon/proc/should_be_critical(new_stat)
+	var/answer = FALSE
+	if(in_shock)
+		answer = TRUE
+	if(health <= hardcrit_threshold)
+		answer = TRUE
+	if(new_stat == SOFT_CRIT || new_stat == HARD_CRIT)
+		answer = TRUE
+	return answer
```

Comment:
set_stat checks for should_be_critical before being changed, this essentially means you can never get out of crit

Reply:
yea that is intended